pyuvdata/uvflag/uvflag.py

```diff
@@ -20,6 +20,40 @@
 __all__ = ["UVFlag", "flags2waterfall", "and_rows_cols", "lst_from_uv"]
 
 
+def _read_uvflag_string(dataset, filename):
+    """
+    Handle backwards compatibility of string stypes for legacy UVFlag files.
+
+    Parameters
+    ----------
+    dataset : h5py dataset
+        The dataset containing string-like data.
+    filename : str
+        The name of the file being read.
+
+    Returns
+    -------
+    str
+        The string from the dataset as a `str` type.
+
+    Notes
+    -----
+    This function is only designed to work on scalar datasets. Arrays of strings
+    should be handled differently.
+
+    """
+    if dataset.dtype.type is np.object_:
+        warnings.warn(
+            f"Strings in metadata of {filename} are not the correct type; "
+            "rewrite with `UVFlag.write` to ensure future compatibility. "
+            "Suppoort for reading these files will be removed in version 2.2.",
```

Comment:
typo: Suppoort -> Support

Reply:
look at us using f strings. We python 3.6 now.